R/extraction_operator_linter.R

```diff
@@ -7,53 +7,27 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 extraction_operator_linter <- function() {
+  xpath <- "
+    //OP-DOLLAR[not(preceding-sibling::expr[1]/SYMBOL[text() = 'self' or text() = '.self'])] |
+    //OP-LEFT-BRACKET[not(following-sibling::expr[1]/descendant::*[not(
+      self::expr or self::OP-PLUS or self::NUM_CONST or self::STR_CONST or self::NULL_CONST
```

Comment:
nit: either put the nodes on new lines or use `xp_or()` -- a bit tough to parse all the nodes in a line visually